### PR TITLE
UI: combine current + history client count tabs into one dashboard

### PR DESCRIPTION
**Pull request by hellobontempo**
### Combines the previous "Current month" and "History" tabs into a single dashboard. 
cc-dashboard

To do (after PR review):
- [x] update file names from 'history' to 'dashboard' (keeping old naming for PR review so the git diff is easier to follow)

### CODE REVIEW

File: ui/.yarn/releases/yarn-1.22.19.js
Comment (hellobontempo) [inline ui/.yarn/releases/yarn-1.22.19.js:71633]: Not really sure how this got in here - need to remove - happened after merging main 🤷

File: ui/.yarn/releases/yarn-1.22.19.js
Comment (hellobontempo) [inline ui/.yarn/releases/yarn-1.22.19.js:71636]: prettier changes

File: ui/app/components/clients/history.js
Comment (hellobontempo) [inline ui/app/components/clients/history.js:9]: leaving this TODO until after PR review so that the git diff is easier to follow

File: ui/app/components/calendar-widget.js
Comment (zofskeez) [inline ui/app/components/calendar-widget.js:20]: Nice work! It looks like the `CalendarWidget` was originally developed specifically for client counts and now with this simplified API it can be used anywhere!

File: ui/.yarn/releases/yarn-1.22.19.js
Comment (hashishaw) [inline ui/.yarn/releases/yarn-1.22.19.js:71633]: According to the [yarn docs,](https://yarnpkg.com/getting-started/qa) this directory should actually be ignored 🤔

File: ui/app/components/calendar-widget.js
Comment (zofskeez) [inline ui/app/components/calendar-widget.js:53]: Since the component is no longer specific to client counts maybe this comment can be tweaked?

File: ui/app/components/calendar-widget.js
Comment (zofskeez) [inline ui/app/components/calendar-widget.js:76]: Might need to confer with design but can this message be made more generic and not reference billing?

File: ui/app/components/clients/history.js
Comment (zofskeez) [inline ui/app/components/clients/history.js:27]: I think these defaults that access `this.args` should be moved to the constructor after `super(...arguments)`. Prior to calling `super` `this.args` would be undefined so I'm surprised if it works like this.

Comment (zofskeez) [reply ui/app/components/clients/history.js:27]: To expand on this a bit further, because we are extending the `Component` class and `args` is likely defined on that class it is accessible since `startMonthTimestamp` is being defined after. I think it's probably a better practice to set class property defaults inside the constructor to ensure that the property is available and has the expected value.

File: ui/app/templates/components/calendar-widget.hbs
Comment (zofskeez) [inline ui/app/templates/components/calendar-widget.hbs:77]: Small nit but this makes it seems like you're calling a function. I think `displayYear` may have more accurately described the property.

File: ui/app/templates/components/calendar-widget.hbs
Comment (zofskeez) [inline ui/app/templates/components/calendar-widget.hbs:108]: Nice I like `month` better than `m`!

**Review by zofskeez — COMMENTED** [review]
Nice work! I'm going to hold off approving until I can test the branch more locally.

File: ui/.yarn/releases/yarn-1.22.19.js
Comment (hellobontempo) [reply ui/.yarn/releases/yarn-1.22.19.js:71633]: ugh yeah... I noticed that. why the heck did this happen then?! 🤔

File: ui/app/components/calendar-widget.js
Comment (hellobontempo) [reply ui/app/components/calendar-widget.js:76]: @ivana-mcconnell - any strong feelings on removing the word 'billing'. It'd make it more sensical for OSS users. 
Screen Shot 2022-11-08 at 11 33 58 AM

File: ui/app/templates/components/calendar-widget.hbs
Comment (hellobontempo) [reply ui/app/templates/components/calendar-widget.hbs:77]: ah yes! I meant to change this back to displayYear - `getDisplayYear` was just supposed to be a temporarily variable while I refactored how they were computed

thank you for calling out!

**Review by zofskeez — APPROVED** [review]

File: ui/app/templates/components/calendar-widget.hbs
Comment (hashishaw) [reply ui/app/templates/components/calendar-widget.hbs:77]: +1

File: ui/app/templates/components/clients/history.hbs
Comment (hashishaw) [inline ui/app/templates/components/clients/history.hbs:62]: This is cleaned up so beautifully ✨

**Review by hashishaw — APPROVED** [review]
Great work cleaning this up!! A couple minor comments but very impressed 🎉

File: ui/app/adapters/clients/activity.js
Comment (hellobontempo) [reply ui/app/adapters/clients/activity.js:9]: Ohh, duh with the or operand `||` instead of the conditional! Good catch

File: ui/app/templates/components/clients/history.hbs
Comment (hellobontempo) [reply ui/app/templates/components/clients/history.hbs:62]: felt so nice to remove all those extra args!

File: ui/app/templates/vault/cluster.hbs
Comment (zofskeez) [inline ui/app/templates/vault/cluster.hbs:83]: I'm curious if this is needed? The component should consider the link active automatically when the url matches the route and `current-when` is typically used to make the link active when it doesn't match.

File: ui/tests/integration/components/date-dropdown-test.js
Comment (zofskeez) [inline ui/tests/integration/components/date-dropdown-test.js:1]: Curious if there is a way to refactor the test(s) in a way where there are not conditional assertions made? Possibly breaking them out into multiple tests, one for each condition.

Comment (hellobontempo) [reply ui/tests/integration/components/date-dropdown-test.js:1]: 🤔 Hmm, the dropdown disables future months and years, depending on the month or year is selected. Either can be selected first, so which items are disabled in the list depends on each selection. 


I think a way I could refactor is pushing all the elements into their respective arrays (i..e `disabledMonths`) and then assert everything in that array is disabled...? But open to ideas

File: ui/tests/integration/components/date-dropdown-test.js
Comment (zofskeez) [inline ui/tests/integration/components/date-dropdown-test.js:180]: Here for example you could remove the first year from that array and move the past year assertions to a specific test. Inside the `yearIdx === 0` block you could do something like this to change the assertion from false to true:
```js
const assertMethod = monthIndex <= CURRENT_MONTH ? 'false' : 'true';
assert[assertMethod](
  month.disabled,
  message
);
```

File: ui/tests/integration/components/date-dropdown-test.js
Comment (zofskeez) [reply ui/tests/integration/components/date-dropdown-test.js:1]: I left a suggestion below for one of the specific cases where you could move the past year assertions to a different test and conditionally use `assert.false` or `assert.true`.

File: ui/app/templates/vault/cluster.hbs
Comment (hellobontempo) [reply ui/app/templates/vault/cluster.hbs:83]: I think you're right. I'll make a ticket to cleanup all of the `@current-when` args, since the other `<LinkTo>` components in this file, and a few elsewhere, still use them.

File: ui/tests/integration/components/date-dropdown-test.js
Comment (hellobontempo) [reply ui/tests/integration/components/date-dropdown-test.js]: I didn't even think to do `assert[method]` - brilliant!

**Review by hashishaw — APPROVED** [review]
Played around with the UI, it's so beautiful! 😭 Great work on this, and the persistence! 💯

File: ui/package.json
Comment (hellobontempo) [reply ui/package.json:117]: this is the component used to render the tooltips in the charts. I updated to see if it fixed the firefox issue - it did not... 😅 haha. But the update included a bug fix, so I thought it'd be good to have

File: ui/app/components/clients/dashboard.js
Comment (hellobontempo) [inline ui/app/components/clients/dashboard.js:8]:
```suggestion
// my sincere apologies to the next dev who has to refactor/debug this (⇀‸↼‶)

**Pull request merged**
